src/appMain/life_cycle.cc

```diff
@@ -381,19 +381,12 @@ void LifeCycle::StopComponents() {
   connection_handler_->Stop();
 
   LOG4CXX_INFO(logger_, "Destroying Protocol Handler");
+  DCHECK_OR_RETURN_VOID(protocol_handler_);
   delete protocol_handler_;
 
   LOG4CXX_INFO(logger_, "Destroying Connection Handler.");
   delete connection_handler_;
 
-#ifdef ENABLE_SECURITY
```

Comment:
@LuxoftAKutsan , this section found when i check this commit. This section creates core dump.